### PR TITLE
campaigns: upsert weebhook only once config is loaded

### PR DESCRIPTION
**Pull request by unknwon**
This PRs correctly reads `externalURL` from site configuration, and blocks until successful read.

This fixes https://github.com/sourcegraph/sourcegraph/issues/8827

### CODE REVIEW

**Comment by unknwon** [comment]
@mrnugget there is an edge case when the admin hasn't configured the `externalURL` in the site config yet. What default value should we use? Or what is expected to happen?

**Comment by mrnugget** [comment]
>  Or what is expected to happen?

I think it should do nothing until that URL is configured. Since `externalURL` is required by now, I think it's fine to depend on that.

File: enterprise/internal/campaigns/webhooks.go [hunk 1]
```diff
@@ -588,6 +588,12 @@ func (*GitHubWebhook) checkRunEvent(cr *gh.CheckRun) *github.CheckRun {
 // Upsert ensures the creation of the BitbucketServer campaigns webhook.
 // This happens periodically at the specified interval.
 func (h *BitbucketServerWebhook) Upsert(every time.Duration) {
+	externalURL := func() string {
```
Comment (tsenart) [inline enterprise/internal/campaigns/webhooks.go]: I was thinking we could apply this to the `globals.ExternalURL` function instead. Any argument against that?

Comment (unknwon) [reply enterprise/internal/campaigns/webhooks.go]: That's the ideal place, see my comments here for why didn't do that: https://github.com/sourcegraph/sourcegraph/issues/8827#issuecomment-596401674

**Comment by unknwon** [comment]
Merging as-is, happy to update logic in the `globals.ExternalURL` if we can make progress on my comment: https://github.com/sourcegraph/sourcegraph/issues/8827#issuecomment-596401674 (i.e. remove CLI flags)

**Pull request merged**